File: Python/231. Power of 2.py
class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n <= 0:
            return False
        while n % 2 == 0:
            n //= 2
        return n == 1
    # Method 2: Bit manipulation (most efficient method)
    # return n > 0 and (n & (n - 1)) == 0

File: JavaScript/231. Power of 2.js
var isPowerOfTwo = function (n) {
  // Method 1: While loop
  if (n <= 0) return false;
  while (n % 2 === 0) {
    n /= 2;
  }
  return n === 1;

  // Method 2: Bit manipulation (most efficient)
  // return n > 0 && (n & (n - 1)) === 0;
};
